src/datasets/infra/repositories/transformers/datasetTransformers.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  Dataset,
+  DatasetVersionState,
+  DatasetMetadataFields,
+  DatasetMetadataBlock,
+  DatasetMetadataSubField,
+  DatasetMetadataFieldValue,
+} from '../../../domain/models/Dataset';
+import { AxiosResponse } from 'axios';
+import { NodeHtmlMarkdown } from 'node-html-markdown';
+
+export const transformVersionResponseToDataset = (response: AxiosResponse): Dataset => {
+  const versionPayload = response.data;
+  return transformVersionPayloadToDataset(versionPayload);
+};
+
+export const transformLatestVersionResponseToDataset = (response: AxiosResponse): Dataset => {
+  const versionPayload = response.data.latestVersion;
+  return transformVersionPayloadToDataset(versionPayload);
+};
+
+const transformVersionPayloadToDataset = (versionPayload: any): Dataset => {
+  return {
+    id: versionPayload.datasetId,
+    persistentId: versionPayload.datasetPersistentId,
+    versionId: versionPayload.id,
+    versionInfo: {
+      majorNumber: versionPayload.versionNumber,
+      minorNumber: versionPayload.versionMinorNumber,
+      state: versionPayload.versionState as DatasetVersionState,
+      createTime: new Date(versionPayload.createTime),
+      lastUpdateTime: new Date(versionPayload.lastUpdateTime),
+      releaseTime: new Date(versionPayload.releaseTime),
+    },
+    license: {
+      name: versionPayload.license.name,
+      uri: versionPayload.license.uri,
+    },
+    metadataBlocks: transformPayloadToDatasetMetadataBlocks(versionPayload.metadataBlocks),
+  };
+};
+
+const transformPayloadToDatasetMetadataBlocks = (metadataBlocksPayload: any): DatasetMetadataBlock[] => {
+  const metadataBlockKeys = Object.keys(metadataBlocksPayload);
+  const datasetMetadataBlocks: DatasetMetadataBlock[] = [];
+  for (let metadataBlockKey of metadataBlockKeys) {
+    datasetMetadataBlocks.push({
+      name: metadataBlockKey,
+      fields: transformPayloadToDatasetMetadataFields(metadataBlocksPayload[metadataBlockKey].fields),
+    });
+  }
+  return datasetMetadataBlocks;
+};
+
+const transformPayloadToDatasetMetadataFields = (metadataFieldsPayload: any): DatasetMetadataFields => {
+  const metadataFieldKeys = Object.keys(metadataFieldsPayload);
+  const metadataFields: DatasetMetadataFields = {};
+  for (let metadataFieldKey of metadataFieldKeys) {
+    const metadataField = metadataFieldsPayload[metadataFieldKey];
+    const metadataFieldTypeName = metadataField.typeName;
+    metadataFields[metadataFieldTypeName] = transformPayloadToDatasetMetadataFieldValue(metadataField.value);
+  }
+  return metadataFields;
+};
+
+const transformPayloadToDatasetMetadataFieldValue = (metadataFieldValuePayload: any): DatasetMetadataFieldValue => {
```

Comment:
I believe you could refactor this whole function using map to improve readability